src/HTTP/HTTP.php

```diff
@@ -193,11 +195,8 @@ public function request_multiple( array $requests ): array {
 	 */
 	private function create_response( int $status, string $body, array $headers ): Response {
 		if (
-			isset( $headers['content-type'] ) && (
-				'application/json' === $headers['content-type'] ||
-				str_starts_with( $headers['content-type'], 'application/json;' ) ||
-				str_starts_with( $headers['content-type'], 'application/json,' )
-			)
+			isset( $headers['content-type'] )
+				&& in_array( 'application/json', array_map( 'trim', explode( ';', $headers['content-type'] ) ) )
```

Comment:
We need to use strict comparisons:
```suggestion
			&& in_array( 'application/json', array_map( 'trim', explode( ';', $headers['content-type'] ) ), true )
```